python/ray/serve/deployment.py

```diff
@@ -297,6 +297,16 @@ def options(
         unchanged from the existing deployment.
         """
         new_config = self._config.copy()
+
+        if num_replicas is not None and _autoscaling_config is not None:
```

Comment:
I don't think we need these checks here. Since `options` is a property of `Deployment`, I think the checks in the `deployment` decorator definition should be sufficient.

Reply:
Resolved asynchronously with @sihanwang41– without this check, users could overwrite their deployment's num_replicas and _autoscaling_config at the same time, which should not be allowed.